src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -926,73 +854,55 @@
                                   "Error: Attempted to configure SSL, but only partial SSL configuration
                                    provided.")))))
 
-(defn get-cn-from-x500-principal
+(schema/defn ^:always-validate get-cn-from-x500-principal :- schema/Str
   "Given an X500Principal object, retrieve the common name (CN)."
-  [x500-principal]
-  {:pre [(x500-principal? x500-principal)]
-   :post [(string? %)]}
+  [x500-principal :- X500Principal]
   (SSLUtils/getCnFromX500Principal x500-principal))
 
-(defn get-cn-from-x509-certificate
+(schema/defn ^:always-validate get-cn-from-x509-certificate :- schema/Str
   "Given an X509Certificate object, retrieve its common name (CN)."
-  [x509-certificate]
-  {:pre [(certificate? x509-certificate)]
-   :post [(string? %)]}
+  [x509-certificate :- X509Certificate]
   (-> (.getSubjectX500Principal x509-certificate)
       get-cn-from-x500-principal))
 
-(defn get-subject-from-x509-certificate
+(schema/defn ^:always-validate get-subject-from-x509-certificate :- schema/Str
   "Given an X509Certificate object, retrieve its subject."
-  [x509-certificate]
-  {:pre [(certificate? x509-certificate)]
-   :post [(string? %)]}
+  [x509-certificate :- X509Certificate]
   (SSLUtils/getSubjectFromX509Certificate x509-certificate))
 
-(defn get-public-key
+(schema/defn ^:always-validate get-public-key :- PublicKey
   "Given an object which contains a public key, extract the public key
   and return it."
-  [key-object]
-  {:pre [(or (certificate-request? key-object)
-             (keypair? key-object))]
-   :post [(public-key? %)]}
+  [key-object :- (schema/cond-pre KeyPair PKCS10CertificationRequest)]
   (SSLUtils/getPublicKey key-object))
 
-(defn get-private-key
+(schema/defn ^:always-validate get-private-key :- PrivateKey
   "Given an object which contains a private key, extract and return it."
-  [key-object]
-  {:pre [(keypair? key-object)]
-   :post [(private-key? %)]}
+  [key-object :- KeyPair]
   (SSLUtils/getPrivateKey key-object))
 
-(defn get-serial
+(schema/defn ^:always-validate get-serial :- schema/Int
```

Comment:
Would it be better to use `BigInteger` here?  `schema/Int` is a bit more broad but probably okay.